Automattic/wp-calypso
# Sign Up: Return to `/sites` when that's the original source

From https://github.com/Automattic/wp-calypso/pull/68545#issuecomment-1267587338

## Proposed Changes

If a user with one site enters the signup flow from `source=sites-dashboard`, always show 'Back to Sites':

<img width="785" alt="image" src="https://user-images.githubusercontent.com/36432/194391837-d5f3f329-a59a-4916-a66d-7cb2c9354499.png">

However, if they enter the signup flow from elsewhere, default to 'Back to My Home':

<img width="701" alt="image" src="https://user-images.githubusercontent.com/36432/194392061-ef5e4cb7-17ac-4641-a0e0-8e2b69e8b2b7.png">

## Testing Instructions

1. Sign in as a WordPress.com user with one site.
2. From My Home Sidebar, click 'Add new site'.
3. Verify the back button appears as 'Back to My Home'.
4. Navigate to `/sites` and click 'Add new site'.
5. Verify the back button appears as 'Back to Sites'.
